src/meta/src/hummock/manager/mod.rs

```diff
@@ -1343,61 +1353,68 @@ impl HummockManager {
-        } else {
-            match compact_task_assignment.remove(task_id) {
-                Some(compact_task) => compact_task.compact_task.unwrap(),
-                None => {
-                    tracing::warn!("{}", format!("compact task {} not found", task_id));
-                    return Ok(false);
-                }
+        // The compaction task is finished.
+        let mut versioning_guard = write_lock!(self, versioning).await;
+        let versioning = versioning_guard.deref_mut();
+        let mut current_version = versioning.current_version.clone();
+        // purge stale compact_status
+        for group_id in original_keys {
+            if !current_version.levels.contains_key(&group_id) {
+                compact_statuses.remove(group_id);
             }
-        };
-
-        {
-            // apply result
-            compact_task.set_task_status(task_status);
-            compact_task.sorted_output_ssts = sorted_output_ssts;
         }
+        let mut tasks = vec![];
 
-        let is_trivial_reclaim = CompactStatus::is_trivial_reclaim(&compact_task);
-        let is_trivial_move = CompactStatus::is_trivial_move_task(&compact_task);
+        let mut hummock_version_deltas = create_trx_wrapper!(
+            self.sql_meta_store(),
+            BTreeMapTransactionWrapper,
+            BTreeMapTransaction::new(&mut versioning.hummock_version_deltas,)
+        );
+        let mut branched_ssts = create_trx_wrapper!(
+            self.sql_meta_store(),
+            BTreeMapTransactionWrapper,
+            BTreeMapTransaction::new(&mut versioning.branched_ssts)
+        );
 
-        {
-            // The compaction task is finished.
-            let mut versioning_guard = write_lock!(self, versioning).await;
-            let versioning = versioning_guard.deref_mut();
-            let mut current_version = versioning.current_version.clone();
-            // purge stale compact_status
-            for group_id in original_keys {
-                if !current_version.levels.contains_key(&group_id) {
-                    compact_statuses.remove(group_id);
+        let mut version_stats = create_trx_wrapper!(
+            self.sql_meta_store(),
+            VarTransactionWrapper,
+            VarTransaction::new(&mut versioning.version_stats)
+        );
+        let mut success_count = 0;
+        let last_version_id = current_version.id;
+        for (idx, task) in report_tasks.into_iter().enumerate() {
+            rets[idx] = true;
+            let mut compact_task = match compact_task_assignment.remove(task.task_id) {
+                Some(compact_task) => compact_task.compact_task.unwrap(),
+                None => {
+                    tracing::warn!("{}", format!("compact task {} not found", task.task_id));
+                    rets[idx] = false;
+                    continue;
                 }
+            };
+
+            {
+                // apply result
+                compact_task.task_status = task.task_status;
```

Comment:
Should we add back this debug assert?
```
debug_assert!(
                compact_task.task_status() != TaskStatus::Pending,
                "report pending compaction task"
            );
```

Reply:
I do not understand, The task shall be 'Pending' before we report it ?